incubator/types-core/src/main/java/io/aklivity/zilla/runtime/types/core/StringValidatorHandler.java

```diff
@@ -38,32 +38,28 @@ public boolean validate(
         int length,
         ValueConsumer next)
     {
-        boolean valid = false;
-
+        boolean valid;
         if ((flags & FLAGS_INIT) != 0x00)
         {
-            initialize = true;
-            incompleteCharCount = 0;
-            incompleteChar = 0;
+            pendingCharBytes = 0;
         }
 
         final int limit = index + length;
         validate:
-        while (index < limit && initialize)
+        while (index < limit)
         {
             final int charByte0 = data.getByte(index);
             final int charByteCount = (charByte0 & 0b1000_0000) != 0
                     ? Integer.numberOfLeadingZeros((~charByte0 & 0xff) << 24)
                     : 1;
```

Comment:
`charByteCount` should move to the `else` clause as it is not needed for the `if` clause.